test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -42,20 +69,23 @@ trait RolesSpec { self: MesosIntTestHelper =>
     assert(m.sparkFramework.isDefined, "The driver should be running")
 
     if (cfg.role != "*") {
-      // TODO: add message
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role))
+
+      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "spark role should exist")
 
       // make sure reserved resources for that role are used
       m.slaves.foreach {
         x =>
           val reserved = x.roleResources.filter(r => r.roleName == cfg.role).head
-          val used = x.usedResources
-          // TODO: add message
-          assert(reserved.resources.cpu >= used.cpu)
+          assert(reserved.resources.cpu == cfg.roleCpus.toInt, "Reserved resources per slave")
+      }
+      val totalCpus = {
+        val tmp = m.slaves.map { x => x.resources.cpu }.sum
```

Comment:
This is the total number of cpu configured on the Mesos cluster. I think it would make more sense to only get the total number of cpus assigned to the role. This number should be exactly the number cpus used by the framework.